### DISCUSSION TITLE
read reference voltage ATTiny202

### DISCUSSION THREAD

**Discussion by jazzjohn27** [SpenceKonde/megaTinyCore#862] [Q&A] [answered]
The following reads the 0.55V internal voltage reference correctly. However if analogReference(INTERNAL2V5) is called to set the 2.5V reference, the reading is 1023. How can the other voltage references be read while using Vdd as the reference?

int x;
void setup(){
 pinMode(0,OUTPUT);
 //analogReference(INTERNAL2V5);
 Serial.begin(9600);
 while(!Serial) ;;;
  }

void loop() 
{
 Serial.print("Reference:   ");
 x =analogRead(ADC_INTREF);
 Serial.println(x);
 delay(1000);   
}

**Comment by jazzjohn27** [comment, marked as answer]
Just found a way by using init_ADC0()

int x;
void setup(){
 pinMode(P2,OUTPUT);
 analogReference(INTERNAL2V5);
 Serial.begin(9600);
 while(!Serial) ;;;
 init_ADC0();  // this makes it work
  }

void loop() 
{
 Serial.print("Reference:   ");
 x =analogRead(ADC_INTREF);
 Serial.println(x );
 delay(1000);   
}

**Reply by SpenceKonde** [reply]
Aaaaah yeah - the trick here is to do 
```
analogReference(INTERNAL2V5); //Sets the reference voltage to 2V5 and sets the ADC to use that reference. 
analogReference(VCC); // sets the ADC to use VDD as reference - but the reference is still set to 2.5V
```

**Reply by SpenceKonde** [reply]
I'll amend Ref_Analog.md
